Find a functionally equivalent code:android.widget.AbsListView.onLayout:COMMENT
Method Modifier: protected   
Comment:/**
 * Subclasses should NOT override this method but
 * {@link #layoutChildren()} instead.
 */

@@ -1,17 +1,18 @@
 {
     super.onLayout(changed, l, t, r, b);
     mInLayout = true;
+    final int childCount = getChildCount();
     if (changed) {
-        int childCount = getChildCount();
         for (int i = 0; i < childCount; i++) {
             getChildAt(i).forceLayout();
         }
         mRecycler.markChildrenDirty();
     }
-    if (mFastScroller != null && (mItemCount != mOldItemCount || mDataChanged)) {
-        mFastScroller.onItemCountChanged(mItemCount);
-    }
     layoutChildren();
     mInLayout = false;
     mOverscrollMax = (b - t) / OVERSCROLL_LIMIT_DIVISOR;
+    // TODO: Move somewhere sane. This doesn't belong in onLayout().
+    if (mFastScroll != null) {
+        mFastScroll.onItemCountChanged(getChildCount(), mItemCount);
+    }
 }

